src/coreclr/vm/frames.cpp

```diff
@@ -2079,6 +2083,40 @@ PCODE UnmanagedToManagedFrame::GetReturnAddress_Impl()
 }
 #endif // FEATURE_COMINTEROP
 
+#ifdef FEATURE_INTERPRETER
+PTR_InterpMethodContextFrame InterpreterEntryFrame::GetInterpMethodTopmostContextFrame()
+{
+    LIMITED_METHOD_CONTRACT;
+    PTR_InterpMethodContextFrame pFrame = m_pInterpMethodContextFrame;
+    _ASSERTE(pFrame != NULL && pFrame->ip != NULL);
+
+    // The frames of a method are linked in a reverse order (from bottom to top of the part of the stack)
```

Comment:
Is there an advantage in this reverse order? I would expect the interpreter frame to point to what we are executing.

Reply:
Vlad was concerned about updating such a pointer on every interpreter to interpreter call. The current call implementation is pretty efficient and according to his experience from Mono, this would add noticeable overhead.

Reply:
Moreover, this reverse walk is only needed in the native debugger on desktop scenario when someone has the debugger sitting in the middle of the `InterpExecMethod`, nothing else needs to walk that way (provided we have the two interpreter frame kinds).

Reply:
> Vlad was concerned about updating such a pointer on every interpreter to interpreter call.

We do not need to update the pointer on every interpreter-to-interpreter call. We only need to update the pointer when we exit the interpreter. ie instead of pushing the full Frame to exit the interpreter, we would just need to patch the current IP in the Interpreter frame.

Reply:
I think that since only the native debugger needs to walk this list and it would need to walk it even if I changed it the way you suggested in case it was sitting in the `InterpExecMethod` and someone has invoked the `clrstack` command, it doesn't seem really beneficial to store it. Unless we moved to the single interpreter frame way, which would introduce the unusual double handling of single explicit frame to the `StackFrameIterator` or something similar to that.
What is your concern about having two separate explicit frames?

Reply:
I think the scheme looks more expensive than it needs to be and non-intuitive (the linked list is in wrong direction).

Reply:
Yes, the direction I am using for this specific native debugger scenario is primarily used for the interpreter internal purposes, I have just reused it for these rare scenarios. This is not used for managed debuggers, GC or EH. In those cases we would always have the interpreter exit frame.